### ISSUE TITLE
Network issues with manual switches

### ISSUE THREAD

**Issue by JammyStuff** [churchers/vm-bhyve#235]
I've noticed some weird network issues with manual switches since upgrading to version 1.2. I'm using iocage and vm-bhyve on the same machine, so I'm using a manual switch to be able to share the same bridge between iocage VNET jails and vm-bhyve VMs.

I have an interface `igb1` with a number of VLAN interfaces (`igb1.100`, `igb1.120`). I also have a bridge `bridge0` which has `igb1.100` as a member, as well as a number of `vnet` members for the iocage jails. I then have a `services` vm-bhyve switch that I created using `vm switch create -t manual -b bridge0 services`. Doing a `vm switch list` yields the following:

```
NAME      TYPE    IFACE    ADDRESS  PRIVATE  MTU  VLAN  PORTS
services  manual  bridge0  n/a      no       n/a  n/a   n/a
```

If I create a VM with an interface on the `services` switch, I'm finding some weird networking behaviour. ping works fine, and if I connect to another server with netcat, everything is fine for both TCP and UDP as long as I don't attempt to send too many characters in a single line. If I attempt to send a long line via TCP, it doesn't send and any future lines sent in the same direction don't arrive. If I attempt to send a long line via UDP, it doesn't send, but subsequent lines sent in the same direction (that are short) arrive fine.

The iocage VNET jails attached to the same bridge don't have the same problem. I also don't remember seeing these problems on pre-1.2 versions of vm-bhyve.

**Comment by churchers** [comment]
Hmm, apart from using groups to identify interfaces instead of the description, which should have no affect on how they function, I can't see that anything is different in recent versions. For manual switches we just assign an id to the bridge, then attach tap interfaces to it.

Sounds like MTU issues but then that would probably need to be very long lines in order to trigger a problem.

It may be worth trying with hardware offload disabled on the physical adaptor to see if that cures it. I've come across several reports of issues that were solved by disabling offload features. I'm sure at least one was on here but I can't find it now. I did find this one, although the problem is not exactly the same as yours. https://forums.freebsd.org/threads/partial-connectivity-issues-from-bhyve-guests-to-jails-and-host.55701/

We shouldn't have to disable these features of course, but I'd try it first just to see if it makes a difference.
